framework/test_selectors/SelectorFactory.cpp

```diff
@@ -188,17 +189,51 @@ struct test *TestrunSelector::testid_to_test(const char *id, bool silent)
     return nullptr;
 }
 
+namespace {
+static constexpr auto arch_spr = "spr";
+static constexpr auto arch_icx = "icx";
+static constexpr auto arch_cpx = "cpx";
+static constexpr auto arch_skx = "skx";
+
+// source https://en.wikichip.org/wiki/intel/cpuid
+const char* get_test_list_name_from_cpu_info(const struct cpu_info& info) {
+    if (info.family != 0x6) {
+        // so far only 0x6 family processors have dedicated test lists
+        return nullptr;
+    }
+
+    if (info.model == 0x8F) {
+        return arch_spr;
+    } else if (info.model == 0x6C || info.model == 0x6A) {
+        return arch_icx;
+    } else if (info.model == 0x55 && info.stepping >= 7) { // TODO stepping: is this true?
+        return arch_cpx;
+    } else if (info.model == 0x55 && info.stepping < 7) { // TODO
+        return arch_skx;
+    } else {
+        return nullptr;
```

Comment:
Please use CPU ISA features to detect the CPU, not the model-stepping.

There's code in `dump_cpu_info()`  in sandstone.cpp to try and detect this, but it's currently broken and only detecting Core2.